esphome/components/online_image/online_image.cpp

```diff
@@ -103,7 +103,16 @@ void OnlineImage::update() {
     ESP_LOGI(TAG, "Updating image");
   }
 
-  this->downloader_ = this->parent_->get(this->url_);
+
+  std::list<http_request::Header> headers;
+  for (const auto &item : this->headers_) {
```

Comment:
Maybe use structured bindings? Should be available from C++17
```suggestion
  for (const auto &[key, value] : this->headers_) {
```
If this works, it would make the rest of the loop slightly better readable (without the ".first()" or ".second()" calls)

Reply:
```
src/esphome/components/online_image/online_image.cpp: In member function 'virtual void esphome::online_image::OnlineImage::update()':
src/esphome/components/online_image/online_image.cpp:107:20: warning: structured bindings only available with -std=c++17 or -std=gnu++17
   for (const auto &[key, value_template] : this->headers_) {
```
Do I set this somewhere or should I revert it?

Reply:
CI doesn't like C++17 so I guess I go back

Reply:
C++11 is the base language level available on all ESPHome supported platforms.

Reply:
Oh sorry. Somehow I thought I had read some discussions, that we were on C++17, and thinking about going to C++20, but looks like I was waaaay off :-(
Might have mistaken it with a different project.